qkernel/makefile

```diff
@@ -24,12 +32,27 @@ $(kernel_debug): kernel_debug $(assembly_object_files)
 	@ld -n --gc-sections -T $(linker_script) -o $(kernel_debug) \
 		$(assembly_object_files) $(qkernel_debug)
 
+$(cc_kernel): cc_kernel $(assembly_object_files)
+	@ld -n --gc-sections -T $(linker_script) -o $(kernel) \
+		$(assembly_object_files) $(qkernel)
+
+$(cc_kernel_debug): cc_kernel_debug $(assembly_object_files)
+	@ld -n --gc-sections -T $(linker_script) -o $(kernel_debug) \
+		$(assembly_object_files) $(qkernel_debug)
+
+
 kernel:
 	CARGO_TARGET_DIR=../target cargo +nightly-2022-08-11-x86_64-unknown-linux-gnu xbuild --target $(arch)-qkernel.json --release
 
 kernel_debug:
 	CARGO_TARGET_DIR=../target cargo +nightly-2022-08-11-x86_64-unknown-linux-gnu xbuild --target $(arch)-qkernel.json
 
+cc_kernel:
+	CARGO_TARGET_DIR=../target cargo +nightly-2022-08-11-x86_64-unknown-linux-gnu xbuild --target $(arch)-qkernel.json --release --features cc
+
+cc_kernel_debug:
```

Comment:
So we are adding two series of targets, all these complexities only too pass the "features += cc" to the cargo build arguments?

I thought it was possible to do that via environment variables but it turns not out to be the case. [This feature request](https://github.com/rust-lang/cargo/issues/4829) has been there for ~7 years, shame.

Reply:
> not good idea, because containerd is not capable to pass extral enviroment variable for cc

You are misunderstanding me. It's only about setting compile-time config. For example to specify a build target you could do `RUSTFLAGS="-C target-cpu=neoverse-n1" cargo build`, this is surprisingly not the case for the `--features` argument.